src/cons/cons.cpp

```diff
@@ -533,8 +536,9 @@ void timewait_stage(const bool reset)
  */
 void apply_ledger(const p2p::proposal &cons_prop)
 {
-    ctx.led_seq_no++;
-    ctx.lcl = cons::save_ledger(cons_prop, ctx.led_seq_no);
+    const std::tuple<const uint64_t, std::string> new_lcl = save_ledger(cons_prop);
```

Comment:
since you are assigning two data types you can use std::pair which is optimised to hold two data types

Reply:
I don't think there's any additional benefits using pair over tuple. In fact tuple gives more flexibility.